app/forms/state_file/id_disability_form.rb

```diff
@@ -27,10 +27,24 @@ def save
       else
         @intake.update(attributes_for(:intake))
       end
+
+      clean_up_followups
     end
 
     private
 
+    def clean_up_followups
+      primary_followups =  @intake.filer_1099_rs(:primary).map(&:state_specific_followup).compact
+      if primary_disabled == "no" || ["spouse", "none"].include?(mfj_disability)
+        primary_followups.each(&:destroy)
+      end
+
+      if @intake.filing_status_mfj? && ["me", "none"].include?(mfj_disability)
+        spouse_followups =  @intake.filer_1099_rs(:spouse).map(&:state_specific_followup).compact
+        spouse_followups.each(&:destroy)
+      end
+    end
+
     def eligible?(primary_or_spouse)
```

Comment:
I have an open question about the `eligible?` method, it looks like we don't update `_disabled` attribute for primary/spouse unless they're `age_eligible`. Not sure why this and `has_taxable_1099r` has anything to do with whether a person is disabled or not?

Reply:
File a bug for this!